File: extract_kafka_batch.py
import os
import sys
PROJECT_ROOT = os.path.dirname(os.path.abspath(__file__))
sys.path.insert(0, PROJECT_ROOT)

from utils.config_loader import load_config
from utils.spark_session import create_spark_session
from extractors.kafka_batch import KafkaBatchExtractor

def main():
    config_path = os.path.join(PROJECT_ROOT, "config.yaml")
    config = load_config(config_path)
    spark = create_spark_session("Kafka Batch Extract")
    extractor = KafkaBatchExtractor(spark, config)
    extractor.extract()
    spark.stop()

if __name__ == "__main__":
    main()

File: ExtractKafkaBatch.scala
/**
 * Batch-экстрактор Kafka.
 *
 * Последовательно выполняет:
 * - создание SparkSession;
 * - чтение сообщений из Kafka;
 * - разбор JSON;
 * - сохранение данных в HDFS.
 */

import reader.KafkaBatchReader
import writer.KafkaBatchWriter

object ExtractKafkaBatch {

  def main(args: Array[String]): Unit = {

    val spark =
      SparkSessionFactory.create("Extract Kafka Batch")

    val transactions =
      new KafkaBatchReader(spark).read()

    new KafkaBatchWriter()
      .write(transactions)

    spark.stop()

  }

}